packages/dev/test/provider/RelayClient.test.ts

```diff
@@ -744,4 +745,67 @@ contract('RelayClient', function (accounts) {
       assert.exists(relayingResult.transaction)
     })
   })
+
+  describe('_resolveConfiguration()', function () {
+    it('should prioritize client config in that order: config function argument, website-supplied config, default config', async function () {
+      sinon.stub(relayClient, '_resolveConfigurationFromServer').returns(Promise.resolve({
+        methodSuffix: 'test suffix from _resolveConfigurationFromServer'
+      }))
+      const config: Partial<GSNConfig> = {
+        methodSuffix: 'test suffix from arg'
+      }
+      let resolvedConfig = await relayClient._resolveConfiguration({
+        provider: relayClient.getUnderlyingProvider(),
+        config
+      })
+      assert.equal(resolvedConfig.methodSuffix, 'test suffix from arg')
+
+      resolvedConfig = await relayClient._resolveConfiguration({
+        provider: relayClient.getUnderlyingProvider(),
+        config: {}
+      })
+      assert.equal(resolvedConfig.methodSuffix, 'test suffix from _resolveConfigurationFromServer')
+
+      sinon.restore()
+      sinon.stub(relayClient, '_resolveConfigurationFromServer').returns(Promise.resolve({}))
+      resolvedConfig = await relayClient._resolveConfiguration({
+        provider: relayClient.getUnderlyingProvider(),
+        config: {}
+      })
+      assert.equal(resolvedConfig.methodSuffix, defaultGsnConfig.methodSuffix)
+      sinon.restore()
+    })
+    it('should not use website configuration if useGsnDocsConfig is false', async function () {
+      const spy = sinon.spy(relayClient, '_resolveConfigurationFromServer')
+      const config: Partial<GSNConfig> = {
+        useGsnDocsConfig: false
```

Comment:
I'm reluctant of leaving the client with silently configuring from a server.
I prefer to add to all our documentation and samples
`{ useDefaultConfig: 'https://config.opengsn.org' }`